service/controller/v19/adapter/guest_subnets.go

```diff
@@ -32,10 +31,39 @@ func (s *GuestSubnetsAdapter) Adapt(cfg Config) error {
 	sort.Slice(zones, func(i, j int) bool {
 		return zones[i].Name < zones[j].Name
 	})
-	for i, az := range zones {
+
+	s.PublicSubnets = []Subnet{
+		Subnet{
+			AvailabilityZone:    zones[0].Name,
+			CIDR:                zones[0].Subnet.Public.CIDR,
```

Comment:
Are we comfortable enough to not verify that `zones` has at least one item?

Reply:
I was thinking the same and initially added checks everywhere. Then lots of tests broke due to only containing relevant data (they were testing different properties) and I considered this second time. If we don't have a single AZ at the moment of CF template creation, something is broken real bad earlier in the chain so I decided to leave them out. I think it's safe to assume that there is at least one AZ when adapter stuff gets called.

Reply:
I agree with you. I added checks where there is direct access to AZs slice. In other places code is in form of
```go
for i := 1; i < key.SpecAvailabilityZones(cfg.CustomObject); i++ { ... }
```

So that cannot break due to invalid indexing even though number of AZs is zero so not worth checking there.